src/loading2/octree-loader.ts

```diff
@@ -415,57 +434,96 @@ export class OctreeLoader {
 	}
 
 	async load(url: string, xhrRequest: XhrRequest) {
-
+		const metadata = await this.fetchMetadata(url, xhrRequest);
+		const attributes = OctreeLoader.parseAttributes(metadata.attributes);
+	
+		this.applyCustomBufferURI(metadata.encoding, attributes);
+	
+		const loader = this.createLoader(url, metadata, attributes);
+	
+		const boundingBox = this.createBoundingBox(metadata);
+		const offset = this.getOffset(boundingBox);
+		const octree = this.initializeOctree(loader, url, metadata, boundingBox, offset, attributes);
+		const root = this.initializeRootNode(octree, boundingBox, metadata);
+		octree.root = root;
+	
+		loader.load(root);
+	
+		return { geometry: octree };
+	}
+	
+	private async fetchMetadata(url: string, xhrRequest: XhrRequest): Promise<Metadata> {
 		const response = await xhrRequest(url);
-		const metadata: Metadata = await response.json();
+		return response.json();
+	}
+	
+	private getBufferUri(attributesObj: any, attributeName: string): string | null {
+		const attribute = attributesObj.attributes.find((attr: any) => attr.name === attributeName);
+		if (attribute) {
+			return attribute.uri;
+		}
+		return null;
+	}
 
-		const attributes = OctreeLoader.parseAttributes(metadata.attributes);
+	private applyCustomBufferURI(encoding: string, attributes: any) {
```

Comment:
This is basically the actual new functionality. Most of the remaining parts of this PR are just refactors.